esrally/mechanic/launcher.py

```diff
@@ -216,14 +148,14 @@ def __init__(self, cfg, metrics_store):
     def start(self, node_configurations):
         nodes = []
         for node_configuration in node_configurations:
-
             node_name = node_configuration.node_name
             host_name = node_configuration.ip
             binary_path = node_configuration.binary_path
             self.binary_paths[node_name] = binary_path
 
-            p = self._start_process(cmd="docker-compose -f %s up" % binary_path, node_name=node_name)
-            # only support a subset of telemetry for Docker hosts (specifically, we do not allow users to enable any devices)
+            p = self._start_process(cmd="docker-compose -f %s up" % binary_path)
```

Comment:
I suggest we modify the `docker-compose.yml` file (btw let's rename it to `.yml.j2`) to have a healthcheck like:

```
    healthcheck:
      test: nc -z 127.0.0.1 {{http_port}}
      interval: 5s
      timeout: 2s
      retries: 10
```

Then we can start in daemonized mode with `-d`  **and** immediately after poll the output of `docker ps -a --filter "id=$(docker-compose -f %s ps -q)" --filter "status=running" --filter "health=healthy" -q` for say 1 minute:

- if result is still an empty line after 1 minute, it failed.
- if successfully up it will return the container id (like: `29f8d70b94dc`), terminate the loop early and return success.

Reply:
I implemented the rough equivalent with the Docker Python SDK.  What do you think?